### ISSUE TITLE
NullSafe issue with kotlin when build freso from source and upload to private maven repo

### ISSUE THREAD

**Issue by glorinli** [facebook/fresco#2589]
### Description

NullSafe issue with kotlin when build freso from source and upload to private maven repo.

### Reproduction

1. Clone fresco, and checkout v2.4.0.
2. Use following command to build and upload
3. Use the build above in my project, some method parameter will become NonNull, while it is Nullable when use official 2.4.0 version.

Build script:
``` bash
rm -rf .gradle
./gradlew clean allclean 
./gradlew build 
./gradlew -P repositoryUrl=[repo_url] uploadArchives
```

Code with issue
``` kotlin
var requestLow: ImageRequest? = ... (so requestLow is nullable)
val controllerBuilder = Fresco.newDraweeControllerBuilder()
            .setOldController(avatarView.controller)
            .setLowResImageRequest(requestLow)
            .setFirstAvailableImageRequests(requests)
            .setAutoPlayAnimations(false)
            .setControllerListener(draweeControllerListener)
```

Above code won't compile, and the error message is:
`Type mismatch: inferred type is ImageRequest? but ImageRequest was expected`

Snapshot:
![image](https://user-images.githubusercontent.com/23448248/110903920-2747da80-8343-11eb-90d3-ef9582771dca.png)

But if I use official fresco 2.4.0 version, everything goes well.

Is there anything I missed when build the fresco?

### Additional Information

* Fresco version: v2.4.0
* Platform version: MacOS 10.15.7
* Jdk: java 11.0.8 2020-07-14 LTS

**Comment by oprisnik** [comment]
Hey! the low res image request is marked to be non-null so this is to be expected. We're using Infer to enforce these now, so the Kotlin code will enforce these and I think this change landed before v2.4.0 was released, so that's interesting.

**Comment by glorinli** [comment]
> Hey! the low res image request is marked to be non-null so this is to be expected. We're using Infer to enforce these now, so the Kotlin code will enforce these and I think this change landed before v2.4.0 was released, so that's interesting.

Yes, I see the infer, but it's weired that if I use official 2.4.0 version, it goes well, and the low res image request parameter is platform types ImageRequest!, I think it's something related to the infer-annotation library, the official build seems contains less dependency of the infer-annotation library.
